test/controllers/SelfAssessmentControllerSpec.scala

```diff
@@ -0,0 +1,176 @@
+      contentAsString(result) should include(messagesApi("label.find_out_how_to_access_your_self_assessment"))
+    }
+  }
+
+  "Calling ApplicationController.ivExemptLandingPage" should {
+    "return 200 for a user who has logged in with GG linked and has a full SA enrollment" in new LocalSetup {
+      override def fakeAuthJourney: FakeAuthJourney = new FakeAuthJourney(ActivatedOnlineFilerSelfAssessmentUser(saUtr))
+
+      val result = controller.ivExemptLandingPage(None)(FakeRequest())
+
+      status(result) shouldBe OK
+    }
+
+    "redirect to the SA activation page on the portal for a user logged in with GG linked to SA which is not yet activated" in new LocalSetup {
+      override def fakeAuthJourney: FakeAuthJourney =
+        new FakeAuthJourney(NotYetActivatedOnlineFilerSelfAssessmentUser(saUtr))
+
+      val result = controller.ivExemptLandingPage(None)(FakeRequest())
+      val doc = Jsoup.parse(contentAsString(result))
+      status(result) shouldBe OK
+
+      doc
+        .getElementsByClass("heading-large")
+        .toString()
+        .contains("Activate your Self Assessment registration") shouldBe true
+    }
+
+    "redirect to 'Find out how to access your Self Assessment' page for a user who has a SAUtr but logged into the wrong GG account" in new LocalSetup {
+      override def fakeAuthJourney: FakeAuthJourney = new FakeAuthJourney(WrongCredentialsSelfAssessmentUser(saUtr))
+
+      val result = controller.ivExemptLandingPage(None)(FakeRequest())
+      val doc = Jsoup.parse(contentAsString(result))
+      status(result) shouldBe SEE_OTHER
+      redirectLocation(result) shouldBe Some(routes.SaWrongCredentialsController.landingPage().url)
+    }
+
+    "render the page for a user who has a SAUtr but has never enrolled" in new LocalSetup {
+      override def fakeAuthJourney: FakeAuthJourney = new FakeAuthJourney(NotEnrolledSelfAssessmentUser(saUtr))
+
+      val result = controller.ivExemptLandingPage(None)(FakeRequest())
+      val doc = Jsoup.parse(contentAsString(result))
+      status(result) shouldBe OK
+
+      doc
+        .getElementsByClass("heading-xlarge")
+        .toString()
+        .contains(messagesApi("label.find_out_how_to_access_your_self_assessment")) shouldBe true
+    }
+
+    "redirect to 'We cannot confirm your identity' page for a user who has no SAUTR" in new LocalSetup {
+      override def fakeAuthJourney: FakeAuthJourney = new FakeAuthJourney(NonFilerSelfAssessmentUser)
+
+      val result = controller.ivExemptLandingPage(None)(FakeRequest())
+      val doc = Jsoup.parse(contentAsString(result))
+      status(result) shouldBe OK
+    }
+
+    "return bad request when continueUrl is not relative" in new LocalSetup {
+      override def fakeAuthJourney: FakeAuthJourney = new FakeAuthJourney(NonFilerSelfAssessmentUser)
+
+      val result = routeWrapper(
+        buildFakeRequestWithAuth("GET", "/personal-account/sa-continue?continueUrl=http://example.com")).get
```

Comment:
Extends `OptionValues` then use `.value` instead of `'.get`. `.value` wll throw a `TestFailedException` instead of a `None.get` exception